### ISSUE TITLE
[core] Complex condition

### ISSUE THREAD

**Issue by decompilercmp** [skylot/jadx#687]
Hello,

In `org/apache/commons/collections4/map/MultiKeyMap:714`
Original Java source code:
```java
public boolean removeAll(Object key1) {
    boolean modified = false;
    final MapIterator<MultiKey<? extends K>, V> it = mapIterator();
    while (it.hasNext()) {
	    final MultiKey<? extends K> multi = it.next();
	    if (multi.size() >= 1 &&
			    (key1 == null ? multi.getKey(0) == null : key1.equals(multi.getKey(0)))) {
		    it.remove();
		    modified = true;
	    }
    }
    return modified;
}
```
if compiled with `javac` and decompiled with jadx 0.9.0

decompiled source code:
```java
public boolean removeAll(Object key1) {
    boolean modified = false;
    MapIterator<MultiKey<? extends K>, V> it = mapIterator();
    while (it.hasNext()) {
        MultiKey<? extends K> multi = (MultiKey) it.next();
        if (multi.size() >= 1) {
            if (key1 == null) {
                if (multi.getKey(0) != null) {
                }
            } else if (!key1.equals(multi.getKey(0))) {
            }
            it.remove();
            modified = true;
        }
    }
    return modified;
}
```

The two statments (`it.remove();` and `modified = true;`) are now executed with only half of the condition beeing true.

**Comment by nitram84** [comment]
This error still happens with latest git version 31d5715

Here is a reduced sample of the original code which causes the same error. I hope it is helpful. Tested with OpenJDK 8 and latest git version. In my opinion a unit test for this case would be a good idea.
 
```java
public class ComplexIf {
	public boolean doSomething(Object obj1, Object obj2) {
	    boolean ret = false;
	    while (!ret) {
		    if (obj1 == null ? obj2 == null : obj1.equals(obj2)) {
			    ret = true;
		    }
	    }
	    return ret;
	}
}
```

**Comment by nitram84** [comment]
Hi @skylot, can you confirm that this issue was fixed as part of the try-catch rewrite? This behaviour change issue was present in b5720bd1 and was fixed at least in eedf32d1. If you can confirm this, this issue can be closed. Can be tested with my sample or with org.apache.commons:commons-collection4:4.1
A unit test for this may still help because I'm pretty sure that this issue is still uncovered and in the same file some regressions with similar cases were introduced.

**Comment by skylot** [comment]
> If you can confirm this, this issue can be closed.

@nitram84 I don't think this is fixed yet. Sure, code duplication produce correct result here, but this is not desired behavior. Also, this is the reason why other similar methods not correct at all. Jadx should correctly restore such conditions  without code duplication. Anyway, thank you for test case and checking my work 😄
